### DISCUSSION TITLE
Need support for DS18B20, Digital Temp sensor

### DISCUSSION THREAD

**Discussion by jarimi** [HomeSpan/HomeSpan#240] [Q&A]
I need help configuring DS18B20 for Homespan, can anyone help?

Sensor uses 1-WIRE PROTOCOL for communication.

I have this working, but need support for homespan:
Text Guide One DS18b20 Sensor: https://randomnerdtutorials.com/esp32-ds18b20-temperature-arduino-ide/

**Comment by HomeSpan** [comment]
Check out my example of a [Temperature Sensor](https://github.com/HomeSpan/TempSensorI2C) using I2C.  You may be able to use the same framework but replace the I2C calls with call to 1-Wire (which needs an external library since 1-Wire is not part of the ESP32 package).  I don't have any one-wire devices, but may buy one to explore and create an example.

**Reply by TomInIowa** [reply]
replace   DEV_Sensors.h with attached file
HomeSpan with DS18B20 Temperature Sensor
[Archive.zip](https://github.com/HomeSpan/HomeSpan/files/8188932/Archive.zip)

**Reply by jaxongatza** [reply]
In DEV_SENSORS.h for //gettemp only keep this.
--------------------------------
double getTemp() {
          sensors.requestTemperatures();
          float temperature_Celsius = sensors.getTempCByIndex(0);


          return(temperature_Celsius);

--------------------------------

The print commands throw an error in the connection process and I'd recommend changing the pin to 2 instead of 21 if you're using esp32.
